src/uu/du/src/du.rs

```diff
@@ -254,35 +255,37 @@ fn unit_string_to_number(s: &str) -> Option<u64> {
     Some(number * multiple.pow(unit))
 }
 
-fn translate_to_pure_number(s: &Option<&str>) -> Option<u64> {
-    match *s {
-        Some(ref s) => unit_string_to_number(s),
-        None => None,
-    }
+fn translate_to_pure_number(s: &str) -> Option<u64> {
+    unit_string_to_number(s)
 }
 
-fn read_block_size(s: Option<&str>) -> u64 {
-    match translate_to_pure_number(&s) {
-        Some(v) => v,
-        None => {
-            if let Some(value) = s {
-                show_error!("invalid --block-size argument '{}'", value);
-            };
-
-            for env_var in &["DU_BLOCK_SIZE", "BLOCK_SIZE", "BLOCKSIZE"] {
-                let env_size = env::var(env_var).ok();
-                if let Some(quantity) = translate_to_pure_number(&env_size.as_deref()) {
-                    return quantity;
-                }
-            }
+fn read_block_size(s: &str) -> Option<u64> {
+    translate_to_pure_number(&s)
+}
+
+fn read_threshold(s: &str) -> (Option<u64>, bool) {
```

Comment:
TODO: Move after get_block_size_from_env (for grouping block size related functions together)